### PR TITLE
Add method: Parse(string input, IFormatProvider provider = null)

### PR DESCRIPTION
**Pull request by ZHZG**
In order to fix issue #15.
We failed to get method Parse with arguments(string and IFormatProvider)
in PropertyTool to parse a string.

So I just add this method and don't use optional arguments for
Parse(string input, IFormatProvider provider, IUnitProvider
unitProvider)

### CODE REVIEW

File: Source/QuantityTypes/Quantities/QuantityTemplate.tt [hunk 1]
```diff
@@ -182,6 +182,31 @@ namespace QuantityTypes
             return value;
         }
 
+		/// <summary>
```
Comment (objorke) [inline Source/QuantityTypes/Quantities/QuantityTemplate.tt]: please fix indentation here!

**Comment by objorke** [comment]
Looks good, but:
1. please add a unit test that demonstrates the problem
2. please create a separate branch/pr for the other defect (and include a test that demonstrates the problem)
3. please add an github issue first, and # reference the issue in the commit description and PR description

File: Source/QuantityTypes.Tests/Quantities/AngleTests.cs [hunk 2]
```diff
@@ -54,5 +54,15 @@ public void ToString_NorwegianCulture_CheckSpaces()
             var a = 90 * Angle.Degree;
             Assert.AreEqual("90,0°", a.ToString("0.0 [°]", CultureInfos.Norwegian));
         }
+
+        [Test]
+        public void ParseStringToAngleType()
```
Comment (objorke) [inline Source/QuantityTypes.Tests/Quantities/AngleTests.cs]: Do we need to use reflection in the test? What do you want to test - getting the correct parse method by reflection, or the parse method itself?

File: Source/QuantityTypes/Quantities/QuantityTemplate.tt [hunk 3]
```diff
@@ -182,6 +182,31 @@ namespace QuantityTypes
             return value;
         }
 
+	/// <summary>
```
Comment (objorke) [inline Source/QuantityTypes/Quantities/QuantityTemplate.tt]: is indentation correct here?

File: Source/QuantityTypes/Quantities/Acceleration.cs [hunk 4]
```diff
@@ -156,6 +156,31 @@ public static Acceleration Parse(string input, IFormatProvider provider = null,
             return value;
         }
 
+		/// <summary>
```
Comment (objorke) [inline Source/QuantityTypes/Quantities/Acceleration.cs]: wrong indentation

File: Source/QuantityTypes/Quantities/Action.cs [hunk 5]
```diff
@@ -142,6 +142,31 @@ public static Action Parse(string input, IFormatProvider provider = null, IUnitP
             return value;
         }
 
+		/// <summary>
```
Comment (objorke) [inline Source/QuantityTypes/Quantities/Action.cs]: indentation looks wrong

**Comment by objorke** [comment]
The code looks ok, but from the web/online diffs the indentation looks wrong. If more commits are needed, please create a new branch and include everything in one commit.

File: Source/QuantityTypes.Tests/Quantities/AngleTests.cs (hunk 2 above)
Comment (ZHZG) [inline Source/QuantityTypes.Tests/Quantities/AngleTests.cs]: Yes, we need use reflection, which is exactly the same as we do in PropertyTools. We want to get the parse method by reflection.

File: Source/QuantityTypes/Quantities/QuantityTemplate.tt (hunk 3 above)
Comment (ZHZG) [inline Source/QuantityTypes/Quantities/QuantityTemplate.tt]: It is correct here.

**Comment by objorke** [comment]
I have started a re-build of this pr

**Comment by ZHZG** [comment]
Just want to say thank you for handling this. Enjoy your vacation.

**Pull request merged**